pkg/workers/move/export.go

```diff
@@ -0,0 +1,393 @@
+	salt := crypto.GenerateRandomBytes(16)
+	createdAt := time.Now()
+
+	exportDoc = &ExportDoc{
+		Domain:       i.Domain,
+		Salt:         salt,
+		State:        ExportStateExporting,
+		CreatedAt:    createdAt,
+		WithDoctypes: opts.WithDoctypes,
+		WithoutIndex: opts.WithoutIndex,
+		TotalSize:    -1,
+	}
+
+	// Cleanup previously archived exports.
+	{
+		var exportedDocs []*ExportDoc
+		exportedDocs, err = GetExports(i.Domain)
+		if err != nil {
+			return
+		}
+		notRemovedDocs := exportedDocs[:0]
+		for _, e := range exportedDocs {
+			if e.State == ExportStateExporting && time.Since(e.CreatedAt) < 24*time.Hour {
+				return nil, ErrExportConflict
+			}
+			notRemovedDocs = append(notRemovedDocs, e)
+		}
+		if len(notRemovedDocs) > 0 {
+			archiver.RemoveArchives(notRemovedDocs)
+		}
+	}
+
+	var size, n int64
+	if err = couchdb.CreateDoc(couchdb.GlobalDB, exportDoc); err != nil {
+		return
+	}
+	realtime.GetHub().Publish(&realtime.Event{
+		Verb:   realtime.EventCreate,
+		Doc:    exportDoc.Clone(),
+		OldDoc: nil,
+		Domain: i.Domain,
+	})
+	defer func() {
+		newExportDoc := exportDoc.Clone().(*ExportDoc)
+		newExportDoc.CreationDuration = time.Since(createdAt)
+		if err == nil {
+			newExportDoc.State = ExportStateDone
+			newExportDoc.TotalSize = size
+		} else {
+			newExportDoc.State = ExportStateError
+			newExportDoc.Error = err.Error()
+		}
+		if erru := couchdb.UpdateDoc(couchdb.GlobalDB, newExportDoc); err == nil {
+			err = erru
+		}
+		realtime.GetHub().Publish(&realtime.Event{
+			Verb:   realtime.EventUpdate,
+			Doc:    newExportDoc.Clone(),
+			OldDoc: exportDoc.Clone(),
+			Domain: i.Domain,
+		})
+	}()
+
+	out, err := archiver.CreateArchive(exportDoc)
+	if err != nil {
+		return
+	}
+	defer func() {
+		if errc := out.Close(); err == nil {
+			err = errc
+		}
+	}()
+
+	gw, err := gzip.NewWriterLevel(out, gzip.BestCompression)
+	if err != nil {
+		return
+	}
+	tw := tar.NewWriter(gw)
+
+	if n, err = writeInstanceDoc(i, "instance", createdAt, tw, nil); err != nil {
+		return
+	}
+	size += n
+
+	settings, err := i.SettingsDocument()
+	if err != nil {
+		return
+	}
+	if n, err = writeDoc("", "settings", settings, createdAt, tw, nil); err != nil {
+		return
+	}
+	size += n
+
+	if !opts.WithoutIndex {
+		var root *vfs.TreeFile
+		root, err = i.VFS().BuildTree()
+		if err != nil {
+			return
+		}
+		n, err = writeDoc("", "files-index", root, createdAt, tw, nil)
+		if err != nil {
+			return
+		}
+		size += n
+
+		cursors, _ := splitFilesIndex(root, nil, BucketSize, BucketSize)
+		exportDoc.IndexFilesCursors = cursors
+	}
+
+	n, err = exportDocs(i, opts.WithDoctypes, createdAt, tw)
+	if errc := tw.Close(); err == nil {
+		err = errc
+	}
+	if errc := gw.Close(); err == nil {
+		err = errc
+	}
+	size += n
+
+	return
+}
+
+// BucketSize is the default size of a file bucket, to split the index into
+// equal-sized parts.
+const BucketSize = 35 * 1024 * 1024 // 50 MB
```

Comment:
Is it 35 or 50 Mo?

Reply:
Thanks for the note, i'll fix it. I'll make it a configurable value I think though.